manifests/package.pp

```diff
@@ -3,4 +3,23 @@
   package { $prosody::package_name:
     ensure  => $prosody::package_ensure,
   }
+
+  if $prosody::manage_repository {
+    $_keyring_path = '/usr/share/keyrings/prosody.gpg'
+
+    apt::source { 'prosody':
+      comment  => 'Prosody',
+      location => "[signed-by=${_keyring_path}] https://packages.prosody.im/debian",
+      release  => $facts['os']['distro']['codename'],
+      repos    => 'main',
```

Comment:
we should also ensure that this repo is configured before the package resource is applied?

Reply:
good catch. fixed now